onnx/onnx.in.proto

```diff
@@ -497,6 +497,50 @@ message TensorShapeProto {
   repeated Dimension dim = 1;
 }
 
+// Sequences
+//
+// Defines a dense, ordered, collection of elements that are of homogeneous types.
+message SequenceProto {
+  // Sequences are made of TypeProto elements, which can either be tensors,
+  // maps, or sequences themselves.
+  message SequenceElement {
+    oneof value {
+      TensorProto tensor_value = 1;
+      SequenceProto sequence_value = 2;
+      MapProto map_value = 3;
+    };
+  };
+  repeated SequenceElement values = 1;
+
+  // shape of the underlying sequence
+  repeated int64 dims = 2;
+
+  // The data type of the sequence.
+  optional TypeProto elem_type = 3;
+}
+
+// Maps
+//
+// Specifies an associative table, defined by a key type and a value type.
+message MapProto {
+  optional int32 key_type = 1;
+  optional int32 key = 2;
+
+  // Map values are made of TypeProto elements, which can either be tensors,
+  // sequences, or maps themselves.
+  message MapElement {
+    oneof value {
+      TensorProto tensor_value = 1;
+      SequenceProto sequence_value = 2;
+      MapProto map_value = 3;
+    };
+  };
+
+  optional TypeProto value_type = 3;
+  repeated MapElement value = 4;
```

Comment:
We need repeated KeyValuePairs, consisting of a key and a value.

Reply:
Added KeyValuePair message type. Thanks for the suggestion.